app/src/main/java/org/oppia/app/player/audio/AudioFragment.kt

```diff
@@ -52,4 +62,25 @@ class AudioFragment : InjectableFragment() {
     languageCodeList.add("hi-en")
     return languageCodeList
   }
+
+  private fun showCellularDataDialogFragment() {
+    cellularDataInterface = object : CellularDataInterface {
```

Comment:
Shouldn't the fragment be implementing this interface? That seems cleaner to me.

Reply:
Unable to understand this point.

Do you mean that `Cellular DataDial Fragment` should implement this interface, if that is the case than I am confused how will be transfer data back to the AudioFragment once user has selected his preference.

Or

Do you mean mean something else?

Reply:
Sorry, what I mean is I think AudioFragment should implement this interface. That way the dialog fragment can do the equivalent:

```kotlin
((CellularDataInterface) getParentFragment()).enableAudioWhileOnCellular(...)
```

This is actually what I thought we were going to do when discussing this yesterday, per: https://stackoverflow.com/a/23144683/3689782.

Reply:
Okay, will change the implementation accordingly.